### PR TITLE
Allow defmt-print to connect to a TCP port.

### PR DESCRIPTION
**Pull request by hydra**
Why: The Segger J-Link debugger sessions listen on TCP port 19021 by default and dump RTT output to any client that connects to the port.

This allows defmt-print to log data so you can see it in real-time WHILE you are debugging in your IDE.

example:
```
$ defmt-print tcp --host=127.0.0.1 --port=19021
```

Output from `--help` is now:

```
defmt-print 0.3.2
Prints defmt-encoded logs to stdout

USAGE:
    defmt-print.exe [FLAGS] -e <elf> [SUBCOMMAND]

FLAGS:
    -h, --help                   Prints help information
        --json
        --show-skipped-frames
    -v, --verbose
    -V, --version

OPTIONS:
    -e <elf>

SUBCOMMANDS:
    help      Prints this message or the help of the given subcommand(s)
    std-in
    tcp
```

Default subcommand is `std-in` for backwards compatibility.

The host and port can read from your environment variables `RTT_HOST` and `RTT_PORT` respectively.

```
$ defmt-print tcp --help
defmt-print-tcp 0.3.2

USAGE:
    defmt-print -e <elf> tcp --host <host> --port <port>

FLAGS:
        --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -h, --host <host>     [env: RTT_HOST=]
    -p, --port <port>     [env: RTT_PORT=]
```

```
C:\> set RTT_HOST=127.0.0.1
C:\> set RTT_PORT=19021
```
or
```
$ export RTT_HOST=127.0.0.1
$ export RTT_PORT=19021
```

Once environment variables are set they are shown and used appropriately.
```
$ defmt-print tcp --help
defmt-print-tcp 0.3.2

USAGE:
    defmt-print -e <elf> tcp --host <host> --port <port>

FLAGS:
        --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -h, --host <host>     [env: RTT_HOST=127.0.0.1]
    -p, --port <port>     [env: RTT_PORT=19021]

$ defmt-print -e <elf> tcp
...
```

J-Link console:
image

CLion debug session of the target:
image

defmt-print output:
image

### CODE REVIEW

**Comment by hydra** [comment]
the 1 failing build is the linter failing on other code in this repo not modified by this PR.

**Comment by Urhengulas** [comment]
https://github.com/knurling-rs/defmt/pull/678 fixes the clippy suggestions. Please rebase your PR after it landed.

File: print/src/main.rs [hunk 1]
```diff
@@ -26,6 +28,28 @@ struct Opts {
...
     #[structopt(short = "V", long)]
     version: bool,
+
+    #[structopt(subcommand)]
+    command: Option<Command>,
+}
+
+#[derive(StructOpt)]
+enum Command {
+    StdIn {
```
Comment (jonas-schievink) [inline print/src/main.rs:38]:
```suggestion
    Stdin {
```

**Comment by Urhengulas** [comment]
Thank you again for your suggestion and for putting in the effort. After further internal discussion we decided to not accept this PR, because it is trivially possible to achieve the same result with a tool like netcat, and we want to keep `defmt-print` simple. 

You can just pipe the data into `defmt-print` like following:
```console
$ nc localhost 19021 | defmt-print -e app-elf
```
Please just comment here if you struggle with getting that to run!

**Comment by hydra** [comment]
@Urhengulas actually that was the first thing I tried, I also tried some other tools and various piping options and so on, but nothing worked with the segger RTT server, hence this PR!  

When piping what happens is that it either a) outputs nothing at all, or b) outputs some log information and then just randomly stops.

I tried the 7.60 and 7.66a Segger RTT client - Windows x10 + Cygwin x64.

Can you please re-open this PR?

File: print/src/main.rs (hunk 1 above)
Comment (hydra) [reply print/src/main.rs:38]: fixed.

**Comment by hydra** [comment]
Rebased on `main` to include #678 and renamed `StdIn` to `Stdin` as per review comments. Also reformatted using `rustfmt`.

**Comment by hydra** [comment]
Today I updated this PR to work with the latest code in `main`.

I still find I need this feature and have used it on a daily basis for the last year now or so now with no issues.

**Comment by Urhengulas** [comment]
I am not able to re-open the PR. Can you please open a new one?

**Comment by hydra** [comment]
> I am not able to re-open the PR. Can you please open a new one?

Done, see #757

**Pull request closed without merging**